server/src/dev/resources/application.yml

```diff
@@ -57,6 +60,43 @@ org:
     miscellaneous:
       allow-anonymous-data-usage: false
 
+bucket4j:
+  enabled: true
+  filters:
+    - cache-name: buckets
+      url: '/api/-/(namespace/create|publish)'
+      http-response-headers:
+        Access-Control-Allow-Origin: '*'
+        Access-Control-Expose-Headers: X-Rate-Limit-Retry-After-Seconds, X-Rate-Limit-Remaining
+      rate-limits:
+        - expression: getParameter("token")
+          bandwidths:
+            - capacity: 15
+              time: 1
+              unit: seconds
+    - cache-name: buckets
+      url: '/vscode/asset/.*/.*/.*/Microsoft.VisualStudio.Services.Icons.Default'
+      http-response-headers:
+        Access-Control-Allow-Origin: '*'
+        Access-Control-Expose-Headers: X-Rate-Limit-Retry-After-Seconds, X-Rate-Limit-Remaining
+      rate-limits:
+        - expression: getRemoteAddr()
+          bandwidths:
+            - capacity: 75
+              time: 1
+              unit: seconds
+    - cache-name: buckets
+      url: '/vscode/(?!asset/.*/.*/.*/Microsoft.VisualStudio.Services.Icons.Default).*|/api/(?!(.*/.*/review(/delete)?)|(-/(namespace/create|publish))).*'
+      http-response-headers:
+        Access-Control-Allow-Origin: '*'
+        Access-Control-Expose-Headers: X-Rate-Limit-Retry-After-Seconds, X-Rate-Limit-Remaining
+      rate-limits:
+        - expression: getRemoteAddr()
+          bandwidths:
+            - capacity: 15
+              time: 1
+              unit: seconds
+
```

Comment:
suggestion (non-blocking): can we add heavier rate-limiting for `/sitemap.xml`? Maybe there are even more endpoints which take a look of compute to respond, but this is one that's really easy to hit and it takes sometimes even 5 seconds to generate (on https://open-vsx.org).

Reply:
Yes, the sitemap endpoint is heavy. It gets all active extensions from the database and turns it into a XML document. It does return caching headers to the client, but it does re-generate the document for every request. So, even if clients respect the caching response headers, the server still re-generates the sitemap for every other client that requests the sitemap. 

I think it's a good idea to cache the result on the server (possibly in combination with ETag headers), so that it only re-generates the sitemap when an extension is added, removed or updated. Or maybe even just generate it once a day.